### ISSUE TITLE
This site can’t be reached ... unexpectedly closed the connection.

### ISSUE THREAD

**Issue by drmrbrewer** [joelgriffith/browserless#9]
I have installed the `browserless/chrome` docker image (as a service), and have opened port 3000, so that it should in theory be accessible via `browserless.mysite.com:3000` (like lots of other web apps I've deployed in the same way).  But when I browse to `browserless.mysite.com:3000` I just get:

    This site can’t be reached
    browserless.mysite.com unexpectedly closed the connection

Any ideas?

**Comment by joelgriffith** [comment]
Can you provide a bit more detail on how you're starting the docker image + how you're connecting to it? Likely there's something not lining up properly:

```
$ docker pull browserless/chrome:latest
$ docker run \
  -d \
  -p 3000:3000 \
  --shm-size 2gb \
  --name browserless \
  --restart always \
  -e "DEBUG=browserless/chrome" \
  -e "CONNECTION_TIMEOUT=300000" \
  -e "MAX_CONCURRENT_SESSIONS=10" \
  -e "MAX_QUEUE_LENGTH=20" \
  browserless/chrome:latest
```

app.js:

```
const puppeteer = require('puppeteer');

(async () => {
    const browser = await puppeteer.connect({
      browserWSEndpoint: 'ws://localhost:3000'
    });

    const page = await browser.newPage();
    await page.goto('https://duckduckgo.com');
    await page.screenshot({ path: './temp.png' });
    browser.close();
})();
```

**Comment by drmrbrewer** [comment]
Doing the `docker run` I get:

    docker: Error response from daemon: driver failed programming external connectivity on endpoint browserless (8d422b4b8eb53fe0cedde86fbee7b044421e2fe0460edb62138454e65813a0d4): Error starting userland proxy: listen tcp 0.0.0.0:3000: bind: address already in use

**Comment by joelgriffith** [comment]
> bind: address already in use
That seems interesting... is there a container on that same machine listening on `3000`?

**Comment by drmrbrewer** [comment]
Yes, indeed there is.  I am actually using [this](https://github.com/githubsaturn/captainduckduck) to deploy my docker-based apps.  And the web console for that is set up on port `3000`.

I assume that a custom port can be specified for browserless, via an environment variable perhaps?  Because when I deploy using the above tool, I don't actually ever call `docker run` and instead it deploys the app as service, based on what is basically a dockerfile along the following lines:

    FROM browserless/chrome
    EXPOSE 3000

The dockerfile is wrapped up in json format, as a [definition file](https://github.com/githubsaturn/captainduckduck/wiki/Captain-Definition-File).  

How would I request a custom port to be used, in the above scenario?

**Comment by joelgriffith** [comment]
Yes, it's possible to change the port mapping: https://docs.browserless.io/docs/docker.html#port
